src/Http/Client/Response.php

```diff
@@ -95,7 +95,7 @@ class Response extends Message implements ResponseInterface
     /**
      * Cookie Collection instance
      *
-     * @var \Cake\Http\Cookie\CookieCollection
+     * @var \Cake\Http\Cookie\CookieCollection|\Cake\Http\Cookie\Cookie[]
```

Comment:
Hmm.. how does it end up being an array?

Reply:
It is not, that means, that it is a collection of Cookie objects, it was only an array if the first class typehint wasnt there.
Same goes for `\ArrayObject|Type[]` etc

Reply:
Okay, though that's a strange way to denote the type of elements the collection contains.

Reply:
Afaik it is the only way for this kind of php magic of hidden iterating over an object.

Reply:
If phpstan/pslam are happy with this then meh.

Reply:
psalm doesn't like the `|\Cake\Http\Cookie\Cookie[]` you have added. Needs to check it's doc if it supports a more sensible way of denoting the elements the collection contains.